lib/eventFactory.js

```diff
@@ -2,9 +2,200 @@
+    const { jobs, nextJobs, prNum } = config;
+    // Get jobsToStart
+    // To start all existing and ENABLED PR-prNum jobs
+    const PRJobArray = jobs.filter(j => j.name.startsWith(`PR-${prNum}:`));
+    const jobsToStart = PRJobArray.filter(j => j.state === 'ENABLED');
+    // Get jobsToCreate
+    // Get all the missing PR- job names
+    const existingPRJobNames = PRJobArray.map(p => p.name);
+    const missingPRJobNames = nextJobs.filter(j => !existingPRJobNames.includes(j));
+    // Get the job name part, e.g. main from PR-1:main
+    const missingJobNames = missingPRJobNames.map(name => name.split(':')[1]);
+    const missingJobs = jobs.filter(j => missingJobNames.includes(j.name));
+
+    return {
+        jobsToStart,
+        jobsToCreate: missingJobs
+    };
+}
+
+/**
+ * Get commit jobs to start that are enabled
+ * @method getJobsFromCommit
+ * @param  {Object}   config
+ * @param  {Array}    config.jobs      Array of job objects
+ * @param  {Object}   config.pipeline  Pipeline object
+ * @param  {String}   config.startFrom Startfrom (e.g. ~commit, ~pr, etc)
+ * @return {Array}                     Array of commit jobs to start
+ */
+function getJobsFromCommit(config) {
+    const { jobs, pipeline, startFrom } = config;
+
+    if (startFrom !== '~commit') {
+        return [];
+    }
+
+    return pipeline.sync()
+    .then((p) => {
+        const nextJobs = workflowParser.getNextJobs(p.workflowGraph, {
+            trigger: startFrom
+        });
+
+        return jobs.filter(j => nextJobs.includes(j.name) && j.state === 'ENABLED');
+    });
+}
+
+/**
+ * Get jobs from job name to start that are enabled
+ * @method getJobsToStart
+ * @param  {Object}   config
+ * @param  {Array}    config.jobs      Array of job objects
+ * @param  {String}   config.startFrom Startfrom (e.g. ~commit, ~pr, etc)
+ * @return {Array}                     Array of jobs to start
+ */
+function getJobsFromJobName(config) {
+    const { jobs, startFrom } = config;
+
+    if (startFrom === '~commit' || startFrom === '~pr') {
+        return [];
+    }
+
+    return jobs.filter(j => j.name === startFrom && j.state === 'ENABLED');
+}
+
+/**
+ * Get PR jobs to start that are enabled
+ * @method getJobsFromPR
+ * @param  {Object}   config
+ * @param  {Object}   config.jobFactory Job Factory
+ * @param  {Array}    config.jobs       Array of job objects
+ * @param  {Object}   config.pipeline   Pipeline object
+ * @param  {Number}   config.prNum      PR number
+ * @param  {String}   config.startFrom  Startfrom (e.g. ~commit, ~pr, etc)
+ * @resolves {Array}                      Array of PR jobs to start
+ */
+function getJobsFromPR(config) {
+    const { jobFactory, jobs, pipeline, prNum, startFrom } = config;
+
+    if (startFrom !== '~pr') {
+        return [];
+    }
+
+    return pipeline.syncPR(prNum)
+    .then(() => {
+        // Get next jobs for when startFrom is ~pr
+        const nextJobs = workflowParser.getNextJobs(pipeline.workflowGraph, {
```

Comment:
This is not right. It should use the `config.workflow` from `pipeline.getConfiguration(config.prRef)` instead of the one from pipeline.